wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java

```diff
@@ -319,7 +332,12 @@ public void prepareRun(StageSubmitterContext context) throws Exception {
     // Parse the recipe and extract all the instances of directives
     // to be processed for extracting lineage.
     RecipeParser recipe = getRecipeParser(context);
-    List<Directive> directives = recipe.parse();
+    List<Directive> directives;
+    try {
+      directives = recipe.parse();
+    } catch (RecipeException e) {
+      throw WranglerErrorUtil.getProgramFailureExceptionDetailsFromChain(e, null, ErrorType.USER);
```

Comment:
Please also take `errorReason` as parameter in the function.

`String errorReason = Unable to parse recipe and extract all instances of directives`.